src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs

```diff
@@ -266,139 +275,187 @@ public void CheckKernel(ContextInformation context, ChainedBlock pindexPrev, uin
                 this.logger.LogTrace("(-)[BAD_STAKE_BLOCK]");
                 ConsensusErrors.BadStakeBlock.Throw();
             }
 
-            pBlockTime = prevBlock.Header.Time;
+            prevBlockTime = prevBlock.Header.Time;
 
-            this.CheckStakeKernelHash(context, pindexPrev, nBits, prevBlock.Header.Time, prevBlockStake, prevUtxo, prevout, (uint)nTime);
+            this.CheckStakeKernelHash(context, prevChainedBlock, headerBits, prevBlock.Header.Time, prevBlockStake, prevUtxo, prevout, (uint)transactionTime);
 
-            this.logger.LogTrace("(-):{0}={1}", nameof(pBlockTime), pBlockTime);
+            this.logger.LogTrace("(-):{0}={1}", nameof(prevBlockTime), prevBlockTime);
         }
 
-        public static ChainedBlock GetLastBlockIndex(StakeChain stakeChain, ChainedBlock index, bool proofOfStake)
+        /// <summary>
+        /// Converts <see cref="BigInteger" /> to <see cref="uint256" />.
+        /// </summary>
+        /// <param name="input"><see cref="BigInteger"/> input value.</param>
+        /// <returns><see cref="uint256"/> version of <paramref name="input"/>.</returns>
+        private uint256 ToUInt256(BigInteger input)
         {
-            if (index == null)
-                throw new ArgumentNullException(nameof(index));
-
-            clogger.LogTrace("({0}:'{1}',{2}:{3})", nameof(index), index, nameof(proofOfStake), proofOfStake);
-
-            BlockStake blockStake = stakeChain.Get(index.HashBlock);
+            byte[] array = input.ToByteArray();
 
-            while ((index.Previous != null) && (blockStake.IsProofOfStake() != proofOfStake))
+            int missingZero = 32 - array.Length;
+            if (missingZero < 0)
             {
-                index = index.Previous;
-                blockStake = stakeChain.Get(index.HashBlock);
+                //throw new InvalidOperationException("Awful bug, this should never happen");
+                array = array.Skip(Math.Abs(missingZero)).ToArray();
             }
 
-            clogger.LogTrace("(-)':{0}'", index);
-            return index;
+            if (missingZero > 0)
+                array = new byte[missingZero].Concat(array).ToArray();
+
+            return new uint256(array, false);
         }
 
         /// <summary>
-        /// Calculates the difficulty target for the next block.
+        /// Converts <see cref="uint256" /> to <see cref="BigInteger" />.
+        /// </summary>
+        /// <param name="input"><see cref="uint256"/> input value.</param>
+        /// <returns><see cref="BigInteger"/> version of <paramref name="input"/>.</returns>
+        private BigInteger FromUInt256(uint256 input)
+        {
+            return BigInteger.Zero;
+        }
+
+        /// <summary>
+        /// Returns <c>true</c> if provided coins were confirmed in less then <paramref name="targetDepth"/> number of blocks.
+        /// </summary>
+        /// <param name="coins">Set of outputs inside a single transaction.</param>
```

Comment:
although this is true, i'd go with `Coins to check confirmation depth for.` because it seems more useful for the caller